mpmath/ctx_fp.py

```diff
@@ -200,7 +203,7 @@ def hypsum(ctx, p, q, types, coeffs, z, maxterms=6000, **kwargs):
             for i in num: t *= (coeffs[i]+k)
             for i in den: t /= (coeffs[i]+k)
             k += 1; t /= k; t *= z; s += t
-            if abs(t) < tol:
+            if not abs(t) >= tol:
```

Comment:
Why is this needed?

Reply:
Yes.

Reply:
BTW, maybe we should explicitly test for nan's (as b0b5446 does), not as here or in 0d97f93.  Or redo the while statement like 064e940.

Reply:
Please check for NaN explicitly, for code readability.

Reply:
Ok, a quick check for nan was added to the hyper(), everything else was reverted.

Reply:
@dimpase, does it make sense for you now?

Reply:
it's still puzzling how it's possible to get to the NaN check without getting an overflow (and thus an uncaught error). I.e., why is this check  needed, at all?

Also, you can get an `inf` rather than NaN, and `inf` is perfectly possible to meaningfully compare with non-infs.

Reply:
> it's still puzzling how it's possible to get to the NaN check without getting an overflow (and thus an uncaught error). I.e., why is this check needed, at all?

The idea was to use double negation (`<` -> `not >=`) to make a quick exit for nan's.

I agree, this is much more clear in form like `while abs(t) >= tol: ...`

> Also, you can get an inf rather than NaN, and inf is perfectly possible to meaningfully compare with non-infs.

I think this piece of code can't handle inf's correctly, so nothing was broken.

Anyway, I've reverted this stuff: an explicit check for nan's in hyper() does the job.  The second commit is optional now, but I think it's a good idea to have same output for fp.mag() and mp.mag().   Right now we have 0 and nan for nan's, respectively.